biorxiv/crawl_extract_metadata_csv.py

```diff
@@ -0,0 +1,155 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# DISCLAIMER: This software is provided "as is" without any warranty,
+# express or implied, including but not limited to the warranties of
+# merchantability, fitness for a particular purpose, and non-infringement.
+#
+# In no event shall the authors or copyright holders be liable for any
+# claim, damages, or other liability, whether in an action of contract,
+# tort, or otherwise, arising from, out of, or in connection with the
+# software or the use or other dealings in the software.
+# -----------------------------------------------------------------------------
+
+# @Author  : Tek Raj Chhetri
+# @Email   : tekraj@mit.edu
+# @Web     : https://tekrajchhetri.com/
+# @File    : crawl_extract_metadata_csv.py
+# @Software: PyCharm
+
+
+import requests
+from bs4 import BeautifulSoup
+import csv
+import random
+import time
+import json
+from urllib.parse import urljoin
+from concurrent.futures import ThreadPoolExecutor, as_completed
+import sys
+def get_user_agents(filepath="user_agents.json"):
+    with open(filepath, "r") as f:
+        try:
+            data = json.load(f)
+            return data["user_agents"]
+        except json.JSONDecodeError as e:
+            sys.stdout.write(f"Error decoding JSON: {e}")
+            return None
+
+
+# Function to get a random User-Agent
+def get_random_user_agent():
+    return random.choice(get_user_agents())
+
+
+# Function to extract information from an article
+def extract_article_info(article_link):
+    headers = {"User-Agent": get_random_user_agent()}
+    try:
+        article_response = requests.get(article_link, headers=headers)
+        article_response.raise_for_status()
+    except requests.RequestException as e:
+        sys.stdout.write(f"Failed to retrieve article page: {article_link} due to: {e}")
+        return None
```

Comment:
Similar to https://github.com/sensein/crawler_publications/pull/1/files#r1644435435